apis/rbac/v1alpha1/types.go

```diff
@@ -0,0 +1,302 @@
+)
+
+// ApplicationParameters parameters for an application.
+type ApplicationParameters struct {
+	// AvailableToOtherTenants - Whether the application is available to other tenants.
+	// +optional
+	// +immutable
+	AvailableToOtherTenants *bool `json:"availableToOtherTenants,omitempty"`
+
+	// DisplayName - The display name of the application.
+	// +optional
+	// +immutable
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Homepage - The home page of the application.
+	// +optional
+	// +immutable
+	Homepage *string `json:"homepage,omitempty"`
+
+	// IdentifierUris - A collection of URIs for the application.
+	// +optional
+	// +immutable
+	IdentifierUris *[]string `json:"identifierUris,omitempty"`
+}
+
+// An ApplicationSpec defines the desired state of an Application.
+type ApplicationSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ApplicationParameters parameters for an application.
+	ApplicationParameters `json:",inline"`
+}
+
+// An ApplicationStatus represents the observed state of an Application.
+type ApplicationStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+
+	// ApplicationID - The application ID.
+	ApplicationID string `json:"applicationID,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// A Application is a managed resource that represents an Application.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type Application struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ApplicationSpec   `json:"spec"`
+	Status ApplicationStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ApplicationList contains a list of Application items
+type ApplicationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Application `json:"items"`
+}
+
+// An ServicePrincipalSpec defines the desired state of an ServicePrincipal.
+type ServicePrincipalSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ApplicationID - The application ID.
+	// +optional
+	ApplicationID string `json:"applicationID,omitempty"`
+
+	// ApplicationIDRef - A reference to the Application id.
+	// +optional
+	// +immutable
+	ApplicationIDRef *xpv1.Reference `json:"applicationIDRef,omitempty"`
+
+	// ApplicationIDSelector - Select a reference to the Application id.
+	// +immutable
+	ApplicationIDSelector *xpv1.Selector `json:"applicationIDSelector,omitempty"`
+
+	// AccountEnabled - whether or not the service principal account is enabled
+	// +optional
+	// +immutable
+	AccountEnabled *bool `json:"accountEnabled,omitempty"`
```

Comment:
Could you please introduce a `ServicePrincipalParameters` struct containing these fields? Also per Crossplane [API conventions](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md), this field should be under `spec.forProvider` and not inlined.  For an example, please see:
https://github.com/crossplane/provider-azure/blob/1c0702110ed45feabe2af15a0478e753d1c34d4e/apis/database/v1beta1/configuration_types.go#L103